File: RUST/Cap07/c07ex01.rs
// c07ex01.rs

use std::io::{self, Write};

fn rotsoma() {
  println!();
  println!("Rotina de Adicao");
  println!();

  print!("Entre o 1o. valor: ");
  io::stdout().flush().unwrap();
  let mut entra = String::new();
  io::stdin().read_line(&mut entra).unwrap();
  let a1: f64 = entra.trim().parse().unwrap();

  print!("Entre o 2o. valor: ");
  io::stdout().flush().unwrap();
  entra.clear();
  io::stdin().read_line(&mut entra).unwrap();
  let b1: f64 = entra.trim().parse().unwrap();

  println!();
  let r1 = a1 + b1;
  println!("O resultado da operacao equivale a: {:.2}", r1);
  println!();
}

fn rotsubtracao() {
  println!();
  println!("Rotina de Subtracao");
  println!();

  print!("Entre o 1o. valor: ");
  io::stdout().flush().unwrap();
  let mut entra = String::new();
  io::stdin().read_line(&mut entra).unwrap();
  let a2: f64 = entra.trim().parse().unwrap();

  print!("Entre o 2o. valor: ");
  io::stdout().flush().unwrap();
  entra.clear();
  io::stdin().read_line(&mut entra).unwrap();
  let b2: f64 = entra.trim().parse().unwrap();

  println!();
  let r2 = a2 - b2;
  println!("O resultado da operacao equivale a: {:.2}", r2);
  println!();
}

fn rotmultiplicacao() {
  println!();
  println!("Rotina de Multiplicacao");
  println!();

  print!("Entre o 1o. valor: ");
  io::stdout().flush().unwrap();
  let mut entra = String::new();
  io::stdin().read_line(&mut entra).unwrap();
  let a3: f64 = entra.trim().parse().unwrap();

  print!("Entre o 2o. valor: ");
  io::stdout().flush().unwrap();
  entra.clear();
  io::stdin().read_line(&mut entra).unwrap();
  let b3: f64 = entra.trim().parse().unwrap();

  println!();
  let r3 = a3 * b3;
  println!("O resultado da operacao equivale a: {:.2}", r3);
  println!();
}

fn rotdivisao() {
  println!();
  println!("Rotina de Divisao");
  println!();

  print!("Entre o 1o. valor: ");
  io::stdout().flush().unwrap();
  let mut entra = String::new();
  io::stdin().read_line(&mut entra).unwrap();
  let a4: f64 = entra.trim().parse().unwrap();

  print!("Entre o 2o. valor: ");
  io::stdout().flush().unwrap();
  entra.clear();
  io::stdin().read_line(&mut entra).unwrap();
  let b4: f64 = entra.trim().parse().unwrap();

  println!();
  if b4 == 0.0 {
    println!("O resultado da operacao equivale a: ERRO");
  } else {
    let r4 = a4 / b4;
    println!("O resultado da operacao equivale a: {:.2}", r4);
  }
  println!();
}

fn main() {
  let mut opcao: i32;

  loop {
    println!("CALCULADORA - V1");
    println!();
    println!("[1] - Adicao");
    println!("[2] - Subtracao");
    println!("[3] - Multiplicacao");
    println!("[4] - Divisao");
    println!("[5] - Fim de Programa");
    println!();
    print!("Escolha uma opcao: ");
    io::stdout().flush().unwrap();

    let mut entra_op = String::new();
    io::stdin().read_line(&mut entra_op).unwrap();
    opcao = entra_op.trim().parse().unwrap_or(0);

    if opcao != 5 {
      match opcao {
        1 => rotsoma(),
        2 => rotsubtracao(),
        3 => rotmultiplicacao(),
        4 => rotdivisao(),
        _ => {
          println!();
          println!("Opcao invalida - Tente novamente");
          println!();
        }
      }
    } else {
      break;
    }
  }
}

File: DART/Cap07/c07ex01.dart
// c07ex01.dart

import 'dart:io';

void rotAdicao() {
  print("");
  print("Rotina de Adicao");
  print("");
  stdout.write("Entre o 1o. valor: ");
  double a1 = double.tryParse(stdin.readLineSync() ?? '') ?? 0.0;
  stdout.write("Entre o 2o. valor: ");
  double b1 = double.tryParse(stdin.readLineSync() ?? '') ?? 0.0;
  print("");
  double r1 = a1 + b1;
  print("O resultado da operacao equivale a: $r1");
  print("");
}

void rotSubtracao() {
  print("");
  print("Rotina de Subtracao");
  print("");
  stdout.write("Entre o 1o. valor: ");
  double a2 = double.tryParse(stdin.readLineSync() ?? '') ?? 0.0;
  stdout.write("Entre o 2o. valor: ");
  double b2 = double.tryParse(stdin.readLineSync() ?? '') ?? 0.0;
  print("");
  double r2 = a2 - b2;
  print("O resultado da operacao equivale a: $r2");
  print("");
}

void rotMultiplicacao() {
  print("");
  print("Rotina de Multiplicacao");
  print("");
  stdout.write("Entre o 1o. valor: ");
  double a3 = double.tryParse(stdin.readLineSync() ?? '') ?? 0.0;
  stdout.write("Entre o 2o. valor: ");
  double b3 = double.tryParse(stdin.readLineSync() ?? '') ?? 0.0;
  print("");
  double r3 = a3 * b3;
  print("O resultado da operacao equivale a: $r3");
  print("");
}

void rotDivisao() {
  print("");
  print("Rotina de Divisao");
  print("");
  stdout.write("Entre o 1o. valor: ");
  double a4 = double.tryParse(stdin.readLineSync() ?? '') ?? 0.0;
  stdout.write("Entre o 2o. valor: ");
  double b4 = double.tryParse(stdin.readLineSync() ?? '') ?? 0.0;
  print("");
  if (b4 == 0.0) {
    print("O resultado da operacao equivale a: ERRO");
  } else {
    double r4 = a4 / b4;
    print("O resultado da operacao equivale a: $r4");
  }
  print("");
}

void main() {
  int opcao = 0;
  while (opcao != 5) {
    print("CALCULADORA - V1");
    print("");
    print("[1] - Adicao");
    print("[2] - Subtracao");
    print("[3] - Multiplicacao");
    print("[4] - Divisao");
    print("[5] - Fim de Programa");
    print("");
    stdout.write("Escolha uma opcao: ");
    opcao = int.tryParse(stdin.readLineSync() ?? '') ?? 0;

    if (opcao != 5) {
      switch (opcao) {
        case 1: rotAdicao();        break;
        case 2: rotSubtracao();     break;
        case 3: rotMultiplicacao(); break;
        case 4: rotDivisao();       break;
        default:
          print("");
          print("Opcao invalida - Tente novamente");
          print("");
          break;
      }
    }
  }
}
